pkg/utils/utils.go

```diff
@@ -0,0 +1,405 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	apis "github.com/openebs/cstor-csi/pkg/apis/cstor/v1"
+	"github.com/openebs/cstor-csi/pkg/cstor/snapshot"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"k8s.io/utils/mount"
+)
+
+const (
+	// TODO make VolumeWaitTimeout as env
+
+	// VolumeWaitTimeout indicates the timegap between two consecutive volume
+	// status check attempts
+	VolumeWaitTimeout = 2
+
+	// TODO make VolumeWaitRetryCount as env
+
+	// VolumeWaitRetryCount indicates the number of retries made to check the
+	// status of volume before erroring out
+	VolumeWaitRetryCount = 6
+
+	// TODO make MonitorMountRetryTimeout as env
+
+	// MonitorMountRetryTimeout indicates the time gap between two consecutive
+	//monitoring attempts
+	MonitorMountRetryTimeout = 5
+)
+
+var (
+	// OpenEBSNamespace is openebs system namespace
+	OpenEBSNamespace string
+
+	// NodeIDENV is the NodeID of the node on which the pod is present
+	NodeIDENV string
+
+	// TransitionVolList contains the list of volumes under transition
+	// This list is protected by TransitionVolListLock
+	TransitionVolList map[string]apis.CSIVolumeStatus
+
+	// TransitionVolListLock is required to protect the above Volumes list
+	TransitionVolListLock sync.RWMutex
+
+	// ReqMountList contains the list of volumes which are required
+	// to be remounted. This list is secured by ReqMountListLock
+	ReqMountList map[string]apis.CSIVolumeStatus
+)
+
+const (
+	// timmeout indiactes the REST call timeouts
+	timeout = 60 * time.Second
+)
+
+func init() {
+
+	OpenEBSNamespace = os.Getenv("OPENEBS_NAMESPACE")
+	if OpenEBSNamespace == "" {
+		logrus.Fatalf("OPENEBS_NAMESPACE environment variable not set")
+	}
+	NodeIDENV = os.Getenv("OPENEBS_NODE_ID")
+	if NodeIDENV == "" && os.Getenv("OPENEBS_NODE_DRIVER") != "" {
+		logrus.Fatalf("OPENEBS_NODE_ID not set")
+	}
+
+	TransitionVolList = make(map[string]apis.CSIVolumeStatus)
+	ReqMountList = make(map[string]apis.CSIVolumeStatus)
+
+}
+
+// parseEndpoint should have a valid prefix(unix/tcp)
+// to return a valid endpoint parts
+func parseEndpoint(ep string) (string, string, error) {
+	if strings.HasPrefix(strings.ToLower(ep), "unix://") ||
+		strings.HasPrefix(strings.ToLower(ep), "tcp://") {
+		s := strings.SplitN(ep, "://", 2)
+		if s[1] != "" {
+			return s[0], s[1], nil
+		}
+	}
+	return "", "", fmt.Errorf("Invalid endpoint: %v", ep)
+}
+
+// logGRPC logs all the grpc related errors, i.e the final errors
+// which are returned to the grpc clients
+func logGRPC(
+	ctx context.Context, req interface{},
+	info *grpc.UnaryServerInfo, handler grpc.UnaryHandler,
+) (interface{}, error) {
+	logrus.Infof("GRPC call: %s", info.FullMethod)
+	logrus.Infof("GRPC request: %s", protosanitizer.StripSecrets(req))
```

Comment:
Since we already have added logs in all the requests, changing this to debugf would be good IMHO, because it will pollute with the get stats logs.